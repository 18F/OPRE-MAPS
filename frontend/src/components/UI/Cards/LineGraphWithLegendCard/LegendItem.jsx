import { faCircle } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import CurrencyFormat from "react-currency-format";
import Tag from "../../Tag";

/**
 * @typedef {Object} LegendItemProps
 * @property {number} activeId - The active id.
 * @property {number} id - The id.
 * @property {string} label - The label.
 * @property {number} value - The value.
 * @property {string} color - The color.
 * @property {string} percent - The percent.
 * @property {string} tagStyleActive - The tag style active.
 */

/**
 * @component LegendItem
 * @param {LegendItemProps} props
 * @returns {JSX.Element}
 */
const LegendItem = ({ activeId, id, label, value, color, percent, tagStyleActive }) => {
    const isGraphActive = activeId === id;

    return (
        <div className="grid-row margin-top-2 font-12px">
            <div className="grid-col-7">
                <div className="display-flex flex-align-center">
                    <FontAwesomeIcon
                        icon={faCircle}
                        className={`height-1 width-1 margin-right-05`}
                        style={{ color: color }}
                    />

                    <span className={isGraphActive ? "fake-bold" : ""}>{label}</span>
                </div>
            </div>
            <div className="grid-col-4">
                <CurrencyFormat
                    value={value}
                    displayType={"text"}
                    thousandSeparator={true}
                    prefix={"$ "}
                    renderText={(value) => <span className={isGraphActive ? "fake-bold" : ""}>{value}</span>}
                />
            </div>
            <div className="grid-col-1">
                <Tag
                    tagStyle="darkTextWhiteBackground"
                    text={percent}
                    label={label}
                    active={isGraphActive}
                    tagStyleActive={tagStyleActive}
                />
            </div>
        </div>
    );
};

export default LegendItem;
