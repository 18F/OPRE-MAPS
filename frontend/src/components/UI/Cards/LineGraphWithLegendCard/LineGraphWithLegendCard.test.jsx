import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import LineGraphWithLegendCard from "./LineGraphWithLegendCard";

// Mock the LineGraph component
vi.mock("../../DataViz/LineGraph", () => ({
    default: () => <div data-testid="mock-line-graph">Mock Line Graph</div>
}));

const mockData = [
    {
        id: 1,
        label: "Test Label 1",
        value: 1000,
        color: "#FF0000",
        percent: "50%",
        tagActiveStyle: "active-style-1"
    },
    {
        id: 2,
        label: "Test Label 2",
        value: 2000,
        color: "#00FF00",
        percent: "30%",
        tagActiveStyle: "active-style-2"
    }
];

const defaultProps = {
    data: mockData,
    bigNumber: 3000,
    heading: "Test Heading"
};

describe("LineGraphWithLegendCard", () => {
    it("renders without crashing", () => {
        render(<LineGraphWithLegendCard {...defaultProps} />);
        expect(screen.getByText("Test Heading")).toBeInTheDocument();
    });

    it("renders correct number of legend items", () => {
        render(<LineGraphWithLegendCard {...defaultProps} />);
        mockData.forEach((item) => {
            expect(screen.getByText(item.label)).toBeInTheDocument();
        });
    });

    it("renders with empty data array", () => {
        render(
            <LineGraphWithLegendCard
                data={[]}
                bigNumber={0}
                heading="Empty Test"
            />
        );
        expect(screen.getByText("Empty Test")).toBeInTheDocument();
    });

    it("displays the correct big number", () => {
        render(<LineGraphWithLegendCard {...defaultProps} />);
        // Note: You might need to adjust this based on how your CurrencyCard formats the number
        expect(screen.getByText("$ 3,000")).toBeInTheDocument();
    });

    it("renders LineGraph component", () => {
        render(<LineGraphWithLegendCard {...defaultProps} />);
        expect(screen.getByTestId("mock-line-graph")).toBeInTheDocument();
    });
});

describe("LineGraphWithLegendCard Snapshot", () => {
    it("matches snapshot", () => {
        const { container } = render(<LineGraphWithLegendCard {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });
});
