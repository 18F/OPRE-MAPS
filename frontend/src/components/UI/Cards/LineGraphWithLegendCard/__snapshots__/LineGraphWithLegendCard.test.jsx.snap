// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LineGraphWithLegendCard Snapshot > matches snapshot 1`] = `
<div>
  <div
    class="bg-brand-base-light-variant border-base-light font-family-sans display-flex _container_eae08c display-inline-block"
    data-cy="currency-summary-card"
    style="padding: 20px 30px 30px 30px; height: 232px;"
  >
    <h3
      class="margin-0 margin-bottom-3 font-12px text-base-dark text-normal"
    >
      Test Heading
    </h3>
    <div>
      <div>
        <span
          class="font-sans-xl text-bold margin-bottom-0"
        >
          $ 3,000
        </span>
        <span
          class="font-sans-3xs text-bold margin-bottom-0"
        >
          .
          00
        </span>
      </div>
    </div>
    <div
      class="margin-top-2"
      id="line-graph-card-:r6:"
    >
      <div
        data-testid="mock-line-graph"
      >
        Mock Line Graph
      </div>
    </div>
    <div
      class="grid-row margin-top-2 font-12px"
    >
      <div
        class="grid-col-7"
      >
        <div
          class="display-flex flex-align-center"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle height-1 width-1 margin-right-05"
            data-icon="circle"
            data-prefix="fas"
            data-testid="legend-icon"
            focusable="false"
            role="img"
            style="color: rgb(255, 0, 0);"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
              fill="currentColor"
            />
          </svg>
          <span
            class=""
            data-testid="label-container"
          >
            Test Label 1
          </span>
        </div>
      </div>
      <div
        class="grid-col-4"
      >
        <span
          class=""
          data-testid="value-container"
        >
          $ 1,000
        </span>
      </div>
      <div
        class="grid-col-1"
      >
        <span
          class="font-12px height-205 radius-md bg-white text-brand-neutral-dark  undefined"
          data-testid="legend-tag"
          style="padding: .25em .5em;"
        >
          50%
        </span>
      </div>
    </div>
    <div
      class="grid-row margin-top-2 font-12px"
    >
      <div
        class="grid-col-7"
      >
        <div
          class="display-flex flex-align-center"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle height-1 width-1 margin-right-05"
            data-icon="circle"
            data-prefix="fas"
            data-testid="legend-icon"
            focusable="false"
            role="img"
            style="color: rgb(0, 255, 0);"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
              fill="currentColor"
            />
          </svg>
          <span
            class=""
            data-testid="label-container"
          >
            Test Label 2
          </span>
        </div>
      </div>
      <div
        class="grid-col-4"
      >
        <span
          class=""
          data-testid="value-container"
        >
          $ 2,000
        </span>
      </div>
      <div
        class="grid-col-1"
      >
        <span
          class="font-12px height-205 radius-md bg-white text-brand-neutral-dark  undefined"
          data-testid="legend-tag"
          style="padding: .25em .5em;"
        >
          30%
        </span>
      </div>
    </div>
  </div>
</div>
`;
